File: reasonix_test/lifes_been_good_project/_build_everywhere/lbg_spring/src/main/java/com/androidfreeman/lifesbeengood/model/Course.java
package com.androidfreeman.lifesbeengood.model;

import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;

@Data
@Entity
@Table(name = "courses")
public class Course implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    private String id;
    private String courseName;
    private String teacherProfileId;
    private String termCode;
    private String color;
    private String credits;
    private String notes;
}

File: reasonix_test/lifes_been_good_project/_build_everywhere/lib/models/course.dart
class Course {
  final String id;
  final String courseName;
  final String teacherProfileId;
  final String termCode;
  final String? color;
  final String? credits;
  final String? notes;

  const Course({
    required this.id,
    required this.courseName,
    required this.teacherProfileId,
    required this.termCode,
    this.color,
    this.credits,
    this.notes,
  });

  factory Course.fromJson(Map<String, dynamic> json) {
    return Course(
      id: (json['id'] ?? '').toString(),
      courseName: (json['course_name'] ?? '').toString(),
      teacherProfileId: (json['teacher_profile_id'] ?? '').toString(),
      termCode: (json['term_code'] ?? '').toString(),
      color: json['color']?.toString(),
      credits: json['credits']?.toString(),
      notes: json['notes']?.toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'course_name': courseName,
      'teacher_profile_id': teacherProfileId,
      'term_code': termCode,
      'color': color,
      'credits': credits,
      'notes': notes,
    };
  }
}

